src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -503,28 +570,26 @@ public function duplicateAction(int $productId): Response
      * Toggles product status
      *
      * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", redirectRoute="admin_products_v2_index")
-     * @DemoRestricted(redirectRoute="admin_products_v2_index")
      *
      * @param int $productId
+     * @param int|null $shopId
      *
      * @return JsonResponse
      */
-    public function toggleStatusAction(int $productId): JsonResponse
+    public function toggleStatusAction(int $productId, ?int $shopId): JsonResponse
     {
-        if ($this->isDemoModeEnabled()) {
-            return $this->json([
-                'status' => false,
-                'message' => $this->getDemoModeErrorMessage(),
-            ]);
-        }
-
-        /** @var bool $isEnabled */
-        $isEnabled = $this->getQueryBus()->handle(new GetProductIsEnabled((int) $productId));
+        $shopConstraint = !empty($shopId) ? ShopConstraint::shop($shopId) : ShopConstraint::allShops();
```

Comment:
so this assumes if shop is not passed then apply for all shops?
Maybe we should homogenize in combinations too? (in combination delete actions we use a request param "allShops" to decide this 🤔 )

Reply:
If the possibility is only one shop or all shops (no shop group ID) then yeah we can do the same for combination